src/Services/nssService.ts

```diff
@@ -787,6 +789,41 @@ module StreamStats.Services {
             }
             return result;
         }
+
+        // get region list form nssservices/regions
+        // referenced getFreshDeskArticles from HelpControllers
+        // keys to success: use ng.Ipromise<any> and return this.Execute 
+        public getRegionList(): ng.IPromise<any> {
+
+            var url = configuration.baseurls['NSS'] + configuration.queryparams['NSSRegions'];
+            var request: WiM.Services.Helpers.RequestInfo = new WiM.Services.Helpers.RequestInfo(url, true);
+            
+            return this.Execute(request).then(
+                (response: any) => {
+                    // create array to return
+                    var regionsRaw = []; 
+
+                    // loop through response and add data to regions array
+                    response.data.forEach(function (element) {
+                        regionsRaw.push(element);
+                    });
+
+                    // add filter for unwanted values, additional values can be added with &&
+                    function filterRegions(element, index, array) {
+                        return element.name !== "Undefined";

```

Comment:
Upon further inspection, I see there are several other regions we'll want to exclude. In the future, we will probably return a boolean variable to https://streamstats.usgs.gov/nssservices/regions to signify whether or not the region should be included in the batch processor list. We will make a new issue when that is ready.

Reply:
Maybe we get rid of this filterRegions for now? Then work on the filtering of which regions should be displayed in that new issue when ready.